### PR TITLE
Nc news 11

### PR DESCRIPTION
**Pull request by Lex5mith**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -359,7 +369,71 @@ describe("app.js tests", () => {
...
+        });
+    });
+
+    test("400: responds with a 400 error and message if order given an invalid value", () => {
+      return request(app)
+        .get("/api/articles?topic=mitch&sort_by=votes&order=carrots")
+        .expect(400)
+        .then((response) => {
+          expect(response.body.msg).toEqual(`Order must be ASC or DESC`);
+        });
```
Comment (johnhutcheon) [inline __tests__/app.test.js:436]: Great testing! All avenues covered, well done!

File: models/article-model.js [hunk 2]
```diff
@@ -19,16 +19,30 @@ exports.fetchArticleById = (article_id) => {
...
 };
 
-exports.allArticlesWithCommentCount = () => {
+exports.allArticlesWithCommentCount = (
+  topic,
+  sort_by = "created_at",
+  order = "DESC"
+) => {
+  const queryValues = [];
+
```
Comment (johnhutcheon) [inline models/article-model.js:43]: This is currently open to SQL injection. We need to think about having some sort of code that ensures we can only pass a valid sort_by or order query.

**Review by johnhutcheon — CHANGES_REQUESTED** [review]
Hey Lex, really excellent testing here! Just please see my comments regarding some extra code we need to add in the model. Have a think about how we may do this. Please resubmit once you've added this.

**Comment by Lex5mith** [comment]
I think I've fixed the promise.reject to gaurd against SQL injection. Please review :-)

**Review by Slyose — APPROVED** [review]
Spot on, now I can't sort by `DROP TABLE articles;` 😆 

Merge away!

**Comment by Lex5mith** [comment]
@Slyose maybe I'm just an anarchist and I wanted you to be able to delete the database!

**Pull request merged**